lib/id3c/db/__init__.py

```diff
@@ -312,3 +312,81 @@ def upsert_sample(db: DatabaseSession,
             LOG.info(f"Upserted sample {sample.id} with collection identifier «{sample.collection_identifier}»")
 
     return sample, status
+
+
+def delete_encounter(db: DatabaseSession, encounter_id: int):
+    """
+    Delete an encounter by its `encounter_id`.
+    Will return the number of rows affected by this operation
+    """
+    with db.cursor() as cursor:
+        cursor.execute(f"DELETE FROM warehouse.encounter WHERE encounter_id = {encounter_id}")
+        assert cursor.rowcount < 2, f"More than one encounter associated with {encounter_id}"
+
+    return cursor.rowcount
+
+
+def delete_encounter_location(db: DatabaseSession, encounter_id: int):
+    """
+    Delete an encounter_location by its `encounter_id`.
+    Will return the number of rows affected by this operation
+    """
+    with db.cursor() as cursor:
+        cursor.execute(f"DELETE FROM warehouse.encounter_location WHERE encounter_id = {encounter_id}")
+
+    return cursor.rowcount
+
+
+def delete_individual(db: DatabaseSession, individual_id: int):
+    """
+    Delete an individual by its `encounter_id`.
+    Will return the number of rows affected by this operation
+    """
+    with db.cursor() as cursor:
+        cursor.execute(f"DELETE FROM warehouse.individual WHERE individual_id = {individual_id}")
+
+    return cursor.rowcount
+
+
+def delete_location(db: DatabaseSession, location_id: int):
```

Comment:
It might be nice to be able to pass in a list of `location_ids` and then delete them with a single query.